product-service/src/functions/addProduct/handler.ts

```diff
@@ -0,0 +1,70 @@
+import 'source-map-support/register';
+
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import { v4 } from 'uuid';
+
+import { formatJSONResponse } from '@libs/apiGateway';
+import { middyfy } from '@libs/lambda';
+import { STATUS_CODES } from '@libs/const';
+
+import { addProduct as addProductToDb } from 'src/services/db';
+
+interface Product {
+    id: string;
+    title: string;
+    description: string;
+    img_url: string;
+    price: number;
+}
+
+export const addProduct: APIGatewayProxyHandler = async (event) => {
+    console.log('addProduct lambda is executing', {
+        method: event.httpMethod,
+        pathParameters: event.pathParameters,
+        body: event.body
+    });
+
+    try {
+        const product = (event.body as Object) as Product;
+        product.id = v4();
```

Comment:
generating id inside of server code seems strange to me

Reply:
moved into db